parcels/particlefile.py

```diff
@@ -165,7 +174,7 @@ def write(self, pset, time, sync=True):
                     logger.warning("Option to_write='once' is not fully functional in indexed mode! Particle properties of such variables are not written for newly added particles.")
                 ind = np.arange(pset.size) + self.idx
                 self.id[ind] = np.array([p.id for p in pset])
-                self.time[ind] = time
+                self.time[ind] = np.array([p.time for p in pset])
```

Comment:
This was not necessary before, because all the particles were at endtime, but now they are at the time they were deleted, right?

Reply:
yes, exactly!